### PR TITLE
Fix gzip compressor instead of test

### PR DESCRIPTION
**Pull request by ArneBab**

### CODE REVIEW

File: src/freenet/support/compress/GzipCompressor.java [hunk 1]
```diff
@@ -33,6 +37,17 @@ public Bucket compress(Bucket data, BucketFactory bf, long maxReadLength, long m
 			Closer.close(is);
 			Closer.close(os);
 		}
+		// force OS byte to 0 regardless of Java version (java 16 changed to setting 255 which would break hashes)
+		if (output instanceof ArrayBucket) {
```
Comment (Bombe) [inline src/freenet/support/compress/GzipCompressor.java:41]: I’d very much prefer something that will never fail, for whatever reasons… how about wrapping the `OutputStream` in another `OutputStream` which will override the 10th byte only? 🤔

Comment (ArneBab) [reply src/freenet/support/compress/GzipCompressor.java:41]: I didn’t see how to wrap the stream, so changed the fallback to copy the bucket, fixing the OS byte.

Comment (Bombe) [reply src/freenet/support/compress/GzipCompressor.java:41]: I imagined something like this: https://github.com/Bombe/fred/commit/5250b03abad4db7c536da3286cf167b1b7015370.

Comment (ArneBab) [reply src/freenet/support/compress/GzipCompressor.java]: @Bombe I now switched to your implementation and removed the meddling at the wrong abstraction. Thank you!

Is is good to go with this?

**Review by Bombe — APPROVED** [review]

**Comment by ArneBab** [comment]
@Bombe thank you! This was the fix that still needed to go in, because the bug would have otherwise caused more and more inserts that would not be reproducible on lower Java versions or fixed hyphanet. Though I think that the effect is limited, because few files are inserted as gzip nowadays (LZMA usually wins).

**Pull request merged**